### DISCUSSION TITLE
How to getValues() from disabled field?

### DISCUSSION THREAD

**Discussion by skorphil** [react-hook-form/react-hook-form#11533] [Q&A]
```jsx
<input
  {...register("test", {
    disabled: true
  })}
/>
```

`getValues('test')` returns nothing

Is there a way to get value from disabled field?

**Comment by skorphil** [comment]
`watch` also not working.
What the point to show value inside the disabled input but not showing it with `getValues` or `watch`?

**Comment by sujinleeme** [comment]
You must use `useWatch` or `watch` because `getValues()` will not trigger re-renders or subscribe to input changes. 
However when you invoke the `getValues()` like in the onClick event you can subscribe values. 

```ts
      <button
        onClick={() => {
          console.log(getValues('test'));
        }}
      />
```

**Reply by skorphil** [reply]
Thank you, will try it one more time, but last time I've tried watch, it did not return value from disabled field as well. Seems like all the methods ignore disabled field value

**Comment by skorphil** [comment]
@sujinleeme 
> You must use `useWatch` or `watch` because `getValues()` will not trigger re-renders or subscribe to input changes. However when you invoke the `getValues()` like in the onClick event you can subscribe values.
> 
> ```ts
>       <button
>         onClick={() => {
>           console.log(getValues('test'));
>         }}
>       />
> ```
 Yeah, watch not working. My scenario is - i create field with a default value in disabled state(tu undisable later). While input is disabled, watch or getValues treat input as empty.

The only way i make it work is to disable field with chakra's props, rather than with use-hook-form
```jsx
<Input
        disabled={true} // working
        {...register(`${institutionName}.name`, {
          // disabled: true, // Not working. Treated as empty by `getValues` or `watch`.
        })}
      />
```

**Reply by sujinleeme** [reply]
As stated in the [official doc](https://chakra-ui.com/docs/components/input/props), you might need to give the `isDisabled` prop, however, `disable` also works fine as you mentioned.  Also, it would be good to check the value of`disabled` returns boolean, not undefined or null.

Here is the [example code](https://codesandbox.io/p/sandbox/chakra-ui-input-error-k43qdk?file=%2Fsrc%2FApp.js%3A51%2C11-51%2C37)
The example: When the "First Name" has error, the second  field, "Last Name" will be disabled.

```ts
        <Input
          isDisabled={!!errors.firstName?.message} // Chakra input's prop
          {...register("lastName", {
            required: "This is required.",
            maxLength: {
              value: 5,
              message: "Max length should be less than 5",
            },
          })}
          placeholder="Last Name"
        />
```

**Comment by Moshyfawn** [comment]
RHF follows the native HTML API behaviour and sets disabled fields to `undefined`.

> ["Set disabled to true will lead input value to be undefined and input control to be disabled."](https://react-hook-form.com/docs/useform/register#:~:text=Set%20disabled%20to%20true%20will%20lead%20input%20value%20to%20be%20undefined%20and%20input%20control%20to%20be%20disabled.)
